vng_api_common/middleware.py

```diff
@@ -96,6 +102,116 @@ def has_scopes(self, scopes: Union[Scope, None]) -> bool:
         return scopes.is_contained_in(scopes_provided)
 
 
+class JWTAuth:
+    def __init__(self, encoded: str = None):
+        self.encoded = encoded
+
+    @property
+    def applicaties(self) -> Union[list, None]:
+        if self.client_id is None:
+            return []
+
+        applicaties = self._get_auth()
+
+        if not applicaties:
+            auth_data = self._request_auth()
+            applicaties = self._save_auth(auth_data)
+
+        return applicaties
+
+    def _request_auth(self) -> list:
+        client = AuthorizationsConfig.get_client()
+
+        response = client.list(
+            'applicatie',
+            query_params={'client_ids': self.client_id}
+        )
+        return underscoreize(response['results'])
+
+    def _get_auth(self):
+        return Applicatie.objects.filter(client_ids__contains=[self.client_id])
+
+    @transaction.atomic
+    def _save_auth(self, auth_data):
+        applicaties = []
+
+        for applicatie_data in auth_data:
+            applicatie_serializer = ApplicatieSerializer(data=applicatie_data)
+            applicatie_serializer.is_valid()
+            applicaties.append(applicatie_serializer.save())
+
+        return applicaties
+
+    @cached_property
+    def client_id(self) -> Union[str, None]:
+        if self.encoded is None:
+            return None
+
+        try:
+            client_id = jwt.decode(self.encoded, verify=False)['client_id']
+        except (jwt.DecodeError, KeyError):
+            logger.warning(_('The support of authorization old format will be terminated soon. '
```

Comment:
*The try/except's in this function needs to be looked at as a whole, but giving hints per statement.*

This warnings seems too soon? We don't know yet if the client uses the old format or forgot the `client_id`/`client_identifier` completely.

I guess the the order of validation should be like:

1. jwt decode error -> exception
2. jwt old style check -> warning in logs
3. jwt client_id missing (old and new) -> exception
4. jwt client_id not found -> exception
5. jwt signature check failed -> exception

Reply:
what should be considered as a "jwt old style check " if it should be separated from the next step?

Reply:
Basically `if 'client_identifier' in jwt.get_unverified_header(self.encoded)` -> log warning

Reply:
`try-except` blocks are rearranged